cogs/general/general.py

```diff
@@ -0,0 +1,433 @@
+            msg = "(づ￣ ³￣)づ{} ⊂(´・ω・｀⊂)".format(name)
+        await ctx.send(msg)
+
+    @commands.command()
+    @commands.guild_only()
+    async def userinfo(self, ctx, *, user: discord.Member=None):
+        """Shows users's informations"""
+        author = ctx.message.author
+        guild = ctx.message.guild
+
+        if not user:
+            user = author
+
+        roles = [x.name for x in user.roles if x.name != "@everyone"]
+
+        joined_at = self.fetch_joined_at(user, guild)
+        since_created = (ctx.message.timestamp - user.created_at).days
+        since_joined = (ctx.message.timestamp - joined_at).days
+        user_joined = joined_at.strftime("%d %b %Y %H:%M")
+        user_created = user.created_at.strftime("%d %b %Y %H:%M")
+        member_number = sorted(guild.members,
+                               key=lambda m: m.joined_at).index(user) + 1
+
+        created_on = "{}\n({} days ago)".format(user_created, since_created)
+        joined_on = "{}\n({} days ago)".format(user_joined, since_joined)
+
+        game = "Chilling in {} status".format(user.status)
+
+        if user.game is None:
+            pass
+        elif user.game.url is None:
+            game = "Playing {}".format(user.game)
+        else:
+            game = "Streaming: [{}]({})".format(user.game, user.game.url)
+
+        if roles:
+            roles = sorted(roles, key=[x.name for x in guild.role_hierarchy
+                                       if x.name != "@everyone"].index)
+            roles = ", ".join(roles)
+        else:
+            roles = "None"
+
+        data = discord.Embed(description=game, colour=user.colour)
+        data.add_field(name="Joined Discord on", value=created_on)
+        data.add_field(name="Joined this guild on", value=joined_on)
+        data.add_field(name="Roles", value=roles, inline=False)
+        data.set_footer(text="Member #{} | User ID:{}"
+                             "".format(member_number, user.id))
+
+        name = str(user)
+        name = " ~ ".join((name, user.nick)) if user.nick else name
+
+        if user.avatar_url:
+            data.set_author(name=name, url=user.avatar_url)
+            data.set_thumbnail(url=user.avatar_url)
+        else:
+            data.set_author(name=name)
+
+        try:
+            await ctx.send(embed=data)
+        except discord.HTTPException:
+            await ctx.send("I need the `Embed links` permission "
+                            "to send this")
+
+    @commands.command()
+    @commands.guild_only()
+    async def guildinfo(self, ctx):
+        """Shows guild's informations"""
+        guild = ctx.message.guild
+        online = len([m.status for m in guild.members
+                      if m.status == discord.Status.online or
+                      m.status == discord.Status.idle])
+        total_users = len(guild.members)
+        text_channels = len([x for x in guild.channels
+                             if x.type == discord.ChannelType.text])
```

Comment:
`type` attribute no longer exists. You might have meant `isinstance`.

You can do e.g. `len(guild.text_channels)` to get the text channel count.